src/main/kotlin/org/rust/ide/utils/checkMatch/Pattern.kt

```diff
@@ -45,8 +43,9 @@ data class Pattern(val ty: Ty, val kind: PatternKind) {
                     is TyTuple -> subPatterns.joinToString(", ", "(", ")") { it.text(ctx) }
                     is TyAdt -> {
                         val name = (ty.item as? RsStructItem)?.name ?: (ty.item as? RsEnumVariant)?.name
+                        val escapedName = name.orEmpty().escapeIdentifierIfNeeded()
```

Comment:
Looks like the corresponding test doesn't cover this change
Let's make the following code work
```rust
struct r#type { r#if: bool }

fn foo(s: r#type) {
    match s/*caret*/ {
        r#type { r#if: true } => {}
    }
}
```

Reply:
Ok, I fixed it and added couple of tests. I didn't know that in some cases this intention can expand struct fields